starknet_devnet/contract_wrapper.py

```diff
@@ -101,36 +75,19 @@ async def call(
         """
         Calls the function identified with `entry_point_selector`, potentially passing in `calldata` and `signature`.
         """
-        starknet_state = self.contract.state.copy()
-        contract_address = self.contract.contract_address
-        selector = entry_point_selector
-
-        if isinstance(contract_address, str):
-            contract_address = int(contract_address, 16)
-        assert isinstance(contract_address, int)
-
-        if isinstance(selector, str):
-            selector = get_selector_from_name(selector)
-        assert isinstance(selector, int)
 
-        if signature is None:
-            signature = []
-
-        internal_tx = InternalInvokeFunction.create(
-            contract_address=contract_address,
-            entry_point_selector=selector,
-            entry_point_type=EntryPointType.EXTERNAL,
+        call_info = await self.contract.state.call_raw(
             calldata=calldata,
+            caller_address=caller_address,
+            contract_address=self.contract.contract_address,
             max_fee=max_fee,
+            selector=entry_point_selector,
             signature=signature,
```

Comment:
Should the same optional casting be done here as is done in `invoke`?

Reply:
[036311d](https://github.com/Shard-Labs/starknet-devnet/pull/99/commits/036311d4540beb097fd06f7bb6d9bfc100134af1)